lib/spack/spack/compiler.py

```diff
@@ -261,15 +262,21 @@ def _find_matches_in_path(cls, compiler_names, detect_version, *path):
         prefixes = [''] + cls.prefixes
         suffixes = [''] + cls.suffixes
 
+        def check_cmp_key(check):
+            name = os.path.basename(check[0])
+            idx = compiler_names.index(name)
+            return idx
+
         checks = []
         for directory in path:
+            dir_checks = []
             if not (os.path.isdir(directory) and
                     os.access(directory, os.R_OK | os.X_OK)):
                 continue
 
             files = os.listdir(directory)
             for exe in files:
-                full_path = os.path.join(directory, exe)
+                full_path = join_path(directory, exe)
```

Comment:
I prefer using `os.path.join` internally, `join_path` is more of a shorthand designed for packages. You already need to `import os` anyway, so this just adds a second unnecessary import.

Reply:
Yeah that's a rebase artifact, I'll fix that.